angular-workspace/projects/ni/nimble-angular/src/directives/anchor-base/disableable-router-link-with-href.directive.ts

```diff
@@ -1,14 +1,23 @@
-import { Directive, ElementRef, Injector } from '@angular/core';
+import { Attribute, Directive, ElementRef, Inject, Renderer2 } from '@angular/core';
 import { LocationStrategy } from '@angular/common';
-import { ActivatedRoute, Router, RouterLinkWithHref } from '@angular/router';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RouterLink } from '../../thirdparty/directives/router_link';
 
 /**
  * Base class for Nimble router link directives that go on disableable elements
  */
 @Directive()
-export class DisableableRouterLinkWithHrefDirective<T extends { disabled: boolean }> extends RouterLinkWithHref {
-    public constructor(injector: Injector, private readonly elementRef: ElementRef<T>) {
-        super(injector.get(Router), injector.get(ActivatedRoute), injector.get(LocationStrategy));
+export class DisableableRouterLinkWithHrefDirective<T extends { disabled: boolean }> extends RouterLink {
+    public constructor(
+    @Inject(Router) router: Router,
+        @Inject(ActivatedRoute) route: ActivatedRoute,
+        // eslint-disable-next-line @angular-eslint/no-attribute-decorator
+        @Attribute('tabindex') tabIndexAttribute: string | null | undefined,
```

Comment:
From code inspection, `tabIndexAttribute` won't be used at all in our version of `router_link.ts` because it is only used with `isAnchorElement` is `false`, and we've hardcoded `isAnchorElement` to `true`. If we want to make it really clear that this won't be used, I could pass `undefined` into the `super()` constructor. Do you think that would be preferred?

Reply:
Yea I think I have a small preference for setting it to undefined. Seems like we can remove the eslint disable and  makes it clearer it's not utilized

Reply:
Didn't check all the RouterLink implementations but looks like at least NimbleBreadcrumbItemRouterLinkWithHrefDirective needs the same treatment

Reply:
Ah, good catch. I've now checked for all places we had been creating our own constructor taking in `Attribute('tabindex')` and removed them.